batchACK_test.go

```diff
@@ -0,0 +1,337 @@
+// Copyright 2018 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package ach
+
+import (
+	"log"
+	"testing"
+)
+
+// mockBatchACKHeader creates a ACK batch header
+func mockBatchACKHeader() *BatchHeader {
+	bh := NewBatchHeader()
+	bh.ServiceClassCode = 220
+	bh.StandardEntryClassCode = "ACK"
+	bh.CompanyName = "Your Company, inc"
+	bh.CompanyIdentification = "231380104"
+	bh.CompanyEntryDescription = "Vndr Pay"
+	bh.ODFIIdentification = "23138010"
+	return bh
+}
+
+// mockACKEntryDetail creates a ACK entry detail
+func mockACKEntryDetail() *EntryDetail {
+	entry := NewEntryDetail()
+	entry.TransactionCode = 24
+	entry.SetRDFI("121042882")
+	entry.DFIAccountNumber = "744-5678-99"
+	entry.Amount = 0
+	entry.SetOriginalTraceNumber("121042880000001")
+	entry.SetReceivingCompany("Best Co. #23")
+	entry.SetTraceNumber(mockBatchACKHeader().ODFIIdentification, 1)
+	entry.DiscretionaryData = "S"
+	return entry
+}
+
+// mockBatchACK creates a ACK batch
+func mockBatchACK() *BatchACK {
+	mockBatch := NewBatchACK(mockBatchACKHeader())
+	mockBatch.AddEntry(mockACKEntryDetail())
+	mockBatch.GetEntries()[0].AddAddenda(mockAddenda05())
+	if err := mockBatch.Create(); err != nil {
+		log.Fatal(err)
```

Comment:
This should be `t.Fatal`, right?

Reply:
The way that is currently defined throughout ach it could be a panic(err), but I'd prefer it log.   At some point we could clean up those test still using panic(err)

Reply:
Ah, that's fine. log.Fatal` calls [do break control flow](https://github.com/golang/go/blob/release-branch.go1.11/src/log/log.go#L305) away from the current goroutine anyway and `testing` picks up those to mark as failed.

Reply:
I would prefer in the library that we always return an error and not dictate a logging structure. We can have stronger opinions in our own services that consume the library.

Reply:
I can make it panic(err)

Reply:
This is test code, so why not `t.Fatal`? That doesn't panic -- it just stops the test.

Reply:
gah! Sorry i can't read a code snippet! I thought that was the factory in the batch.go